### ISSUE TITLE
Variables in GraphQL queries?

### ISSUE THREAD

**Issue by kkor** [birkir/gatsby-source-prismic-graphql#5]
Hæ!

So, I'm using templates with graphQL variables in the queries, like in this starter example:
https://github.com/LekoArts/gatsby-starter-prismic/blob/master/src/templates/post.jsx

The query I'm making in the page.tsx template looks like this:

```
export const query = graphql`
    query PageByUid($uid: String!) {
        prismic {
            page(uid: $uid, lang: "en-us") {
                title
            }
        }
    }
`;
```

And that works for the published content, but the preview query fails, because it doesn't add the variables:
![image](https://user-images.githubusercontent.com/1140636/52410032-d6b88000-2adf-11e9-85f3-b3429f1e5a9b.png)

Testing the same query with "query variables" `{
  "uid": "my-first-page"
}` in the Prismic GraphiQL (https://my-repo.prismic.io/graphql) adds them correctly:

![image](https://user-images.githubusercontent.com/1140636/52409813-41b58700-2adf-11e9-88f4-369cfc45815a.png)

a) Am I missing something or is this not supported?
b) If it's not supported: Do you use something else to setup dynamic pages? If there's no workaround, it's a big blocker for me :(
b) If there's no workaround: Is it a big thing to add? Could I help?
c) If it's possible to add: How many other features are missing compared to gatsby-source-prismic? I'm just getting started with gatsby and graphQL, so I don't even know what features I will need, but this project I'm setting up is quite large, so probably a lot of them, whatever they are?

Thanks!

**Comment by birkir** [comment]
This is a great catch and I think I can implement it!

I will also add a function to customize the variables if needed.

👍🏻

**Comment by hjortureh** [comment]
Hæ Birkir, any plans to add this soon? I think getting this to work along with the slices issue is probably important for many ppl in order to use the plugin.

**Comment by birkir** [comment]
Fixed! Now passing pageContext by default in 2.2.0

**Comment by webonmars** [comment]
Hey Guys,
I have exactly the same problem right now...How can I fix it?
